torch/_dynamo/variables/tensor.py

```diff
@@ -252,6 +251,13 @@ def call_method(
         elif name == "size" and self.size is not None:
             sizes = [variables.ConstantVariable(x) for x in self.size]
             constant_result = SizeVariable(sizes, **options)
+
+            if "dim" in kwargs:
+                dim = kwargs.pop("dim")
+                constant_result = constant_result.call_method(
+                    tx, "__getitem__", [dim], {}
```

Comment:
btw I think it would be better to use `getitem_const` similar to what's used for positional dim argument

Reply:
`getitem_const` is better. Safety is not the issue, this code is signaling a `call_method` which is not what we are doing here. `call_method` is not just "Invoke thing on thing", it mimics the fx node op name for a reason, and `call_method` on a VariableTracker is meant to record an operation. This being a constant it's mostly the same, but call_method still incurs a propagation for no reason.

Reply:
Thanks for your review, I thought if we use ```getitem_const``` directly we have to move all the logic inside ```call_method``` there, in case of ```config.dynamic_shapes``` is True for instance (But since this is my first contribution I'm not sure if this happens or not). 

Will open a new pull request  and fix this.

Reply:
Dynamic shapes are handled in the next conditional (and they use `call_method`). Interestingly, there's no special handling for `strides` and `dynamic_shapes` - could it lead to overspecialization?